File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int mx=Integer.MIN_VALUE;
        List<Boolean> ans=new ArrayList<>();
        
        for(int i=0;i<candies.length;i++)
            {
            mx=Math.max(mx,candies[i]);
            }
        
      for(int i=0;i<candies.length;i++)
          {
          if(candies[i]+extraCandies>=mx)
              { ans.add(true);}
          
          else{ ans.add(false);}
          
          }
          
        return ans;
        
            
    }
}

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        
        
      vector<bool>v;
        
     int  mx=*max_element(candies.begin(),candies.end());
     // int mx=INT_MIN;
        
        for(int i=0;i<candies.size();i++)
            {
            //mx=max(mx,candies[i]);
                   if(candies[i]+extraCandies>=mx)
           v.push_back(true);
            
            else{ v.push_back(false);
                   
                   }
            }
                   
           
      return v;             
    }
};